haskell/private/pkgdb_to_bzl.py

```diff
@@ -133,39 +133,30 @@ def hs_library_pattern(name, mode = "static", config = ""):
     # this does not work, we generate a path in the workspace and
     # output a SYMLINK information for the parent process
 
-    symlinks = []
-
     # first, try to get a path within the package
     # We check if the file exists because cabal will unconditionally
     # generate the database entry even if no haddock was generated.
-    if pkg.haddock_html and os.path.exists(pkg.haddock_html):
+    haddock_html = None
+    if pkg.haddock_html:
         haddock_html = path_to_label(pkg.haddock_html, pkg.pkgroot)
-
         if not haddock_html:
             haddock_html = os.path.join("haddock", "html", pkg.name)
-            symlinks.append((pkg.haddock_html, haddock_html))
-    else:
-        haddock_html = None
-
-    haddock_interfaces = []
+            output.append("#SYMLINK: {} {}".format(pkg.haddock_html, haddock_html))
 
     # If there is many interfaces, we give them a number
     interface_id = 0
-
+    haddock_interfaces = []
     for interface_path in pkg.haddock_interfaces:
-        if os.path.exists(interface_path):
```

Comment:
You mostly removed the `os.path.exists` call. Could you elaborate why?

Reply:
Because I believe that is an incorrect test, which does not match the semantics of `path_to_label()`. The latter returns `None` if `$topdir` was not found in the input. In the case of bindists, we do have a `$topdir` to expand, and the given path does not exist unless we expand the path. The path always exists because it's somewhere underneath the `$topdir`.